src/Symfony/Component/Form/ChoiceList/Factory/PropertyAccessDecorator.php

```diff
@@ -172,6 +172,11 @@ public function createView(ChoiceListInterface $list, $preferredChoices = null,
 
         if (is_string($label) && !is_callable($label)) {
             $label = new PropertyPath($label);
+        } elseif (is_string($label) && is_callable($label)) {
```

Comment:
Maybe this should be the responsibility of the `DefaultChoiceListFactory` as the interface is supposed to [support `callable`](https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Form/ChoiceList/Factory/ChoiceListFactoryInterface.php#L84) anyway.

Then it would just need to make `PropertyAccessDecorator` treat callable strings as strings, instead of passing them to the decorated factory as it is.

But one using the `DefaultChoiceListFactory` without decorator should not experience this bug.

What do you think @webmozart ?